operators/pkg/controller/common/defaults/pod_template.go

```diff
@@ -203,22 +203,85 @@ func (b *PodTemplateBuilder) WithTerminationGracePeriod(period int64) *PodTempla
 	return b
 }
 
-// initContainerExists checks if an init container with the given name already exists in the template.
-func (b *PodTemplateBuilder) initContainerExists(name string) bool {
-	for _, c := range b.PodTemplate.Spec.InitContainers {
+// findVolumeMountByNameOrMountPath attempts to find a volume mount with the given name or mount path in the mounts
+// Returns the index of the volume mount or -1 if no volume mount by that name was found.
+func (b *PodTemplateBuilder) findVolumeMountByNameOrMountPath(
+	volumeMount corev1.VolumeMount,
+	mounts []corev1.VolumeMount,
+) int {
+	for i, vm := range mounts {
+		if vm.Name == volumeMount.Name || vm.MountPath == volumeMount.MountPath {
+			return i
+		}
+	}
+	return -1
+}
+
+// WithInitContainerDefaults sets default values for the current init containers.
+//
+// Defaults:
+// - If the init container contains an empty image field, it's inherited from the main container.
+// - VolumeMounts from the main container are added to the init container VolumeMounts, unless they would conflict
+//   with a specified VolumeMount (by having the same VolumeMount.Name or VolumeMount.MountPath)
+func (b *PodTemplateBuilder) WithInitContainerDefaults() *PodTemplateBuilder {
```

Comment:
I think it would be safe to execute this as part of `PodTemplateBuilder.setDefaults()`, to enforce it for ES, Kibana & APM without the need to call it explicitly. What do you think?

Reply:
Unfortunately, this needs to happen /after/ the Volumes and VolumeMounts have been included.